[Bug]: error reading VPC Endpoint Route Table Association (...): couldn't find resource

### Terraform Core Version

1.2.9

### AWS Provider Version

4.3.5

### Affected Resource(s)

* aws_vpc_endpoint_route_table_association

### Expected Behavior

resource should have been created without error

### Actual Behavior

creation failed because resource cannot be found (because of eventual consistency of AWS api) - similarly to https://github.com/hashicorp/terraform-provider-aws/issues/21629

### Relevant Error/Panic Output Snippet

```shell
2023-04-25T18:20:16.365+0200 [ERROR] provider.terraform-provider-aws_v4.35.0_x5: Response contains error diagnostic: tf_proto_version=5.3 tf_req_id=d7450a73-3452-113c-b277-a83f0d44e474 tf_resource_type=aws_vpc_endpoint_route_table_association tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_detail= tf_provider_addr=registry.terraform.io/hashicorp/aws diagnostic_severity=ERROR diagnostic_summary="error reading VPC Endpoint Route Table Association (vpce-0ec749d5e82b5d7f4/rtb-0f5b710b97279ba71): couldn't find resource" timestamp=2023-04-25T18:20:16.363+0200
```


### Terraform Configuration Files

resource "aws_vpc_endpoint_route_table_association" "public_s3" {
  vpc_endpoint_id = aws_vpc_endpoint.s3.id
  route_table_id  = aws_route_table.main.id
}

### Steps to Reproduce

Create the resource in a loop run repeatedly ideally tens or hundreds of times until you hit this AWS race/eventual consistency bug

### Debug Output

```
2023-04-25T18:20:15.444+0200 [TRACE] provider.terraform-provider-aws_v4.35.0_x5: Received request: tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=d7450a73-3452-113c-b277-a83f0d44e474 tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:805 @module=sdk.proto tf_proto_version=5.3 tf_resource_type=aws_vpc_endpoint_route_table_association timestamp=2023-04-25T18:20:15.443+0200
2023-04-25T18:20:15.444+0200 [TRACE] provider.terraform-provider-aws_v4.35.0_x5: Sending request downstream: tf_proto_version=5.3 tf_req_id=d7450a73-3452-113c-b277-a83f0d44e474 tf_resource_type=aws_vpc_endpoint_route_table_association tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:17 @module=sdk.proto tf_provider_addr=registry.terraform.io/hashicorp/aws timestamp=2023-04-25T18:20:15.443+0200
2023-04-25T18:20:15.444+0200 [TRACE] provider.terraform-provider-aws_v4.35.0_x5: calling downstream server: @module=sdk.mux tf_mux_provider=*schema.GRPCProviderServer tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-mux@v0.7.0/internal/logging/mux.go:16 timestamp=2023-04-25T18:20:15.443+0200
2023-04-25T18:20:15.444+0200 [TRACE] provider.terraform-provider-aws_v4.35.0_x5: Calling downstream: @module=sdk.helper_schema tf_mux_provider=*schema.GRPCProviderServer tf_resource_type=aws_vpc_endpoint_route_table_association tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/resource.go:836 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=d7450a73-3452-113c-b277-a83f0d44e474 timestamp=2023-04-25T18:20:15.444+0200
2023-04-25T18:20:15.444+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: [DEBUG] Creating VPC Endpoint Route Table Association: {
2023-04-25T18:20:15.444+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:   AddRouteTableIds: ["rtb-0f5b710b97279ba71"],
2023-04-25T18:20:15.444+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:   VpcEndpointId: "vpce-0ec749d5e82b5d7f4"
2023-04-25T18:20:15.444+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: }


2023-04-25T18:20:15.444+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: [DEBUG] [aws-sdk-go] DEBUG: Request ec2/ModifyVpcEndpoint Details:
2023-04-25T18:20:15.444+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: ---[ REQUEST POST-SIGN ]-----------------------------
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: POST / HTTP/1.1
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Host: ec2.us-east-1.amazonaws.com
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: User-Agent: APN/1.0 HashiCorp/1.0 Terraform/1.2.9 (+https://www.terraform.io) terraform-provider-aws/dev (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.114 (go1.18.4; darwin; arm64) HashiCorp-terraform-exec/0.17.2
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Content-Length: 120
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Authorization: AWS4-HMAC-SHA256 Credential=ASIA57YFSNYF72DLK557/20230425/us-east-1/ec2/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=ac469cd7dd6e63c593697a071dc4fd8b95c40ed9e96f76a603b6ad6581f0d2df
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Content-Type: application/x-www-form-urlencoded; charset=utf-8
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: X-Amz-Date: 20230425T162015Z
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: X-Amz-Security-Token: IQoJb3JpZ2luX2VjEBAaCXVzLWVhc3QtMiJHMEUCICC6ISG3gwqQ1t0C8sdqox6pyEs8pD69QgYMRYc41UVmAiEA3RbvmSgyLKE+aboChXXENF3CXjIoMabcw7E438lUXtgqjgMIGRABGgw5NjE1NDc0OTY5NzEiDOCuAn13BHYbwt3MHyrrAiup5gUJhLPn/2Tpr6JFwgX0cDnjb96/nvbn0zCofA36XeuxleQP7yRbS6etb9LI91IStSIvT0V+us8+Tuw/IchGEKuRMd1MsIhEpXT5kS49c3d7ow5kkaWXpPJZFGMjlW0iZ1H0hmK8fRIKs00Xo9omGtn5TN4G8W6ARYSDA2KTi0Qy0JMKcco+1DG1Sc55kOJBkCMulRbvsmod2J9xh0Xf6Pdf+MHpJwC9077DpJ021KleQZb1Or9erfTsIas5PM0yVFsJ63l1vNHSUKRxu4+ivMSRc+peZhK4D18wJDEUI0Va4XAXto6hb/31YlD5uwA0dWgxthQY0SAzj27DcoRTCNJYho200jg0hjh3s3aXVVRALwsGi2k6BvMQ3gJYXO6SA31LPShQWY2mfK6nE58YOMeuk2GWfHesKkCrthkmaJ4EuYDj2jDZN2L6lHVESOhZ9Yay9VpY03yHYoEGwhc7oKU5qEAa6cHjIzCq+5+iBjqmATdO8n4KdtuR8cgFbNkxWP+GsSajoB+G+TOAnaS9PRmvdc00udTNNW8HRzyPIINHv9BLpHE/kJNTI4C9TX7qnXmC13W1H+AVrU2Lqdk6Utpt4eLoUD2H8+pyiZQHJgg5yF+L3dr2I26N5CyfdbZBEsw1HMGU+jzK5VoY6BcIHHhmCjw9u6/a3zAdXxIiZO+XOlsJb42KC3VwAcZU6Csj5XftcsZUMNM=
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Accept-Encoding: gzip
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: 
2023-04-25T18:20:15.445+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Action=ModifyVpcEndpoint&AddRouteTableId.1=rtb-0f5b710b97279ba71&Version=2016-11-15&VpcEndpointId=vpce-0ec749d5e82b5d7f4

2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: [DEBUG] [aws-sdk-go] <?xml version="1.0" encoding="UTF-8"?>
2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: <ModifyVpcEndpointResponse xmlns="http://ec2.amazonaws.com/doc/2016-11-15/">
2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:     <requestId>4937bcad-d068-4b8a-a2b3-49393d0e8932</requestId>
2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:     <return>true</return>
2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: </ModifyVpcEndpointResponse>


2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: [DEBUG] [aws-sdk-go] DEBUG: Request ec2/DescribeVpcEndpoints Details:
2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: ---[ REQUEST POST-SIGN ]-----------------------------
2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: POST / HTTP/1.1
2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Host: ec2.us-east-1.amazonaws.com
2023-04-25T18:20:15.712+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: User-Agent: APN/1.0 HashiCorp/1.0 Terraform/1.2.9 (+https://www.terraform.io) terraform-provider-aws/dev (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.114 (go1.18.4; darwin; arm64) HashiCorp-terraform-exec/0.17.2
2023-04-25T18:20:15.713+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Content-Length: 85
2023-04-25T18:20:15.713+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Authorization: AWS4-HMAC-SHA256 Credential=ASIA57YFSNYF72DLK557/20230425/us-east-1/ec2/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=88c200a9e19e1df07fc9487d5a8981b9cf1699af76741b5c3b8e4db6dbe27e53
2023-04-25T18:20:15.713+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Content-Type: application/x-www-form-urlencoded; charset=utf-8
2023-04-25T18:20:15.713+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: X-Amz-Date: 20230425T162015Z
2023-04-25T18:20:15.713+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: X-Amz-Security-Token: IQoJb3JpZ2luX2VjEBAaCXVzLWVhc3QtMiJHMEUCICC6ISG3gwqQ1t0C8sdqox6pyEs8pD69QgYMRYc41UVmAiEA3RbvmSgyLKE+aboChXXENF3CXjIoMabcw7E438lUXtgqjgMIGRABGgw5NjE1NDc0OTY5NzEiDOCuAn13BHYbwt3MHyrrAiup5gUJhLPn/2Tpr6JFwgX0cDnjb96/nvbn0zCofA36XeuxleQP7yRbS6etb9LI91IStSIvT0V+us8+Tuw/IchGEKuRMd1MsIhEpXT5kS49c3d7ow5kkaWXpPJZFGMjlW0iZ1H0hmK8fRIKs00Xo9omGtn5TN4G8W6ARYSDA2KTi0Qy0JMKcco+1DG1Sc55kOJBkCMulRbvsmod2J9xh0Xf6Pdf+MHpJwC9077DpJ021KleQZb1Or9erfTsIas5PM0yVFsJ63l1vNHSUKRxu4+ivMSRc+peZhK4D18wJDEUI0Va4XAXto6hb/31YlD5uwA0dWgxthQY0SAzj27DcoRTCNJYho200jg0hjh3s3aXVVRALwsGi2k6BvMQ3gJYXO6SA31LPShQWY2mfK6nE58YOMeuk2GWfHesKkCrthkmaJ4EuYDj2jDZN2L6lHVESOhZ9Yay9VpY03yHYoEGwhc7oKU5qEAa6cHjIzCq+5+iBjqmATdO8n4KdtuR8cgFbNkxWP+GsSajoB+G+TOAnaS9PRmvdc00udTNNW8HRzyPIINHv9BLpHE/kJNTI4C9TX7qnXmC13W1H+AVrU2Lqdk6Utpt4eLoUD2H8+pyiZQHJgg5yF+L3dr2I26N5CyfdbZBEsw1HMGU+jzK5VoY6BcIHHhmCjw9u6/a3zAdXxIiZO+XOlsJb42KC3VwAcZU6Csj5XftcsZUMNM=
2023-04-25T18:20:15.713+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Accept-Encoding: gzip
2023-04-25T18:20:15.713+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: 
2023-04-25T18:20:15.713+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: Action=DescribeVpcEndpoints&Version=2016-11-15&VpcEndpointId.1=vpce-0ec749d5e82b5d7f4

2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: -----------------------------------------------------
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: [DEBUG] [aws-sdk-go] <?xml version="1.0" encoding="UTF-8"?>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: <DescribeVpcEndpointsResponse xmlns="http://ec2.amazonaws.com/doc/2016-11-15/">
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:     <requestId>328edb08-9ba3-4c69-ace1-efc35c9ab5ae</requestId>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:     <vpcEndpointSet>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:         <item>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <creationTimestamp>2023-04-25T16:20:09.000Z</creationTimestamp>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <dnsEntrySet/>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <groupSet/>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <networkInterfaceIdSet/>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <ownerId>961547496971</ownerId>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <policyDocument>{"Version":"2008-10-17","Statement":[{"Effect":"Allow","Principal":"*","Action":"*","Resource":"*"}]}</policyDocument>
2023-04-25T18:20:15.890+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <privateDnsEnabled>false</privateDnsEnabled>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <requesterManaged>false</requesterManaged>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <routeTableIdSet>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                 <item>rtb-0f5b710b97279ba71</item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                 <item>rtb-01e0983f1d3d37234</item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                 <item>rtb-04a8f08ac3b17836a</item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                 <item>rtb-08d46b5d4c94e3082</item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             </routeTableIdSet>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <serviceName>com.amazonaws.us-east-1.s3</serviceName>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <state>available</state>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <subnetIdSet/>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <tagSet>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                 <item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                     <key>redpanda-org</key>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                     <value>alena@redpanda.com</value>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                 </item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                 <item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                     <key>Name</key>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                     <value>network-av0set0tf0log0level0</value>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:                 </item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             </tagSet>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <vpcEndpointId>vpce-0ec749d5e82b5d7f4</vpcEndpointId>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <vpcEndpointType>Gateway</vpcEndpointType>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:             <vpcId>vpc-03920329e35f1345a</vpcId>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:         </item>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5:     </vpcEndpointSet>
2023-04-25T18:20:15.891+0200 [DEBUG] provider.terraform-provider-aws_v4.35.0_x5: </DescribeVpcEndpointsResponse>




2023-04-25T18:20:16.365+0200 [TRACE] provider.terraform-provider-aws_v4.35.0_x5: Called downstream: tf_resource_type=aws_vpc_endpoint_route_table_association @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/resource.go:838 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_mux_provider=*schema.GRPCProviderServer tf_req_id=d7450a73-3452-113c-b277-a83f0d44e474 tf_rpc=ApplyResourceChange @module=sdk.helper_schema timestamp=2023-04-25T18:20:16.363+0200
2023-04-25T18:20:16.365+0200 [TRACE] provider.terraform-provider-aws_v4.35.0_x5: Received downstream response: tf_req_id=d7450a73-3452-113c-b277-a83f0d44e474 tf_resource_type=aws_vpc_endpoint_route_table_association @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:37 diagnostic_error_count=1 diagnostic_warning_count=0 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_duration_ms=919 tf_rpc=ApplyResourceChange @module=sdk.proto tf_proto_version=5.3 timestamp=2023-04-25T18:20:16.363+0200
2023-04-25T18:20:16.365+0200 [ERROR] provider.terraform-provider-aws_v4.35.0_x5: Response contains error diagnostic: tf_proto_version=5.3 tf_req_id=d7450a73-3452-113c-b277-a83f0d44e474 tf_resource_type=aws_vpc_endpoint_route_table_association tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_detail= tf_provider_addr=registry.terraform.io/hashicorp/aws diagnostic_severity=ERROR diagnostic_summary="error reading VPC Endpoint Route Table Association (vpce-0ec749d5e82b5d7f4/rtb-0f5b710b97279ba71): couldn't find resource" timestamp=2023-04-25T18:20:16.363+0200
2023-04-25T18:20:16.365+0200 [TRACE] provider.terraform-provider-aws_v4.35.0_x5: Served request: @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:831 tf_proto_version=5.3 @module=sdk.proto tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=d7450a73-3452-113c-b277-a83f0d44e474 tf_resource_type=aws_vpc_endpoint_route_table_association tf_rpc=ApplyResourceChange timestamp=2023-04-25T18:20:16.363+0200
```

### Panic Output

_No response_

### Important Factoids

As you can see in the trace logs, the repeated describe already returned the existing association but then the next request fails

### References

Similar to https://github.com/hashicorp/terraform-provider-aws/issues/21629

### Would you like to implement a fix?

Yes

--- a/.changelog/30994.txt
+++ b/.changelog/30994.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_vpc_endpoint_route_table_association: Retry resource Create for EC2 eventual consistency
+```

--- a/internal/service/ec2/vpc_endpoint_route_table_association.go
+++ b/internal/service/ec2/vpc_endpoint_route_table_association.go
@@ -57,7 +57,6 @@ func resourceVPCEndpointRouteTableAssociationCreate(ctx context.Context, d *sche
 
 	log.Printf("[DEBUG] Creating VPC Endpoint Route Table Association: %s", input)
 	_, err := conn.ModifyVpcEndpointWithContext(ctx, input)
-
 	if err != nil {
 		return sdkdiag.AppendErrorf(diags, "creating VPC Endpoint Route Table Association (%s): %s", id, err)
 	}
@@ -82,7 +81,9 @@ func resourceVPCEndpointRouteTableAssociationRead(ctx context.Context, d *schema
 	// Human friendly ID for error messages since d.Id() is non-descriptive
 	id := fmt.Sprintf("%s/%s", endpointID, routeTableID)
 
-	err := FindVPCEndpointRouteTableAssociationExists(ctx, conn, endpointID, routeTableID)
+	_, err := tfresource.RetryWhenNewResourceNotFound(ctx, RouteTableAssociationPropagationTimeout, func() (interface{}, error) {
+		return nil, FindVPCEndpointRouteTableAssociationExists(ctx, conn, endpointID, routeTableID)
+	}, d.IsNewResource())
 
 	if !d.IsNewResource() && tfresource.NotFound(err) {
 		log.Printf("[WARN] VPC Endpoint Route Table Association (%s) not found, removing from state", id)

